DSharpPlus.SlashCommands/SlashCommandsExtension.cs

```diff
@@ -444,20 +509,30 @@ void AddContextMenus(IEnumerable<MethodInfo> contextMethods)
         /// <param name="parameters">The parameters to convert.</param>
         /// <param name="guildId">The guild id to register the commands to. <see langword="null"/> to register the slash command globally.</param>
         /// <returns>A list of <see cref="DiscordApplicationCommandOption"/>, options Discord can parse in slash commands.</returns>
+        [SuppressMessage("Styling", "IDE0047", Justification = "Readability for basic arithmetic.")]
         private async Task<List<DiscordApplicationCommandOption>> ParseParameters(ParameterInfo[] parameters, ulong? guildId)
         {
             var options = new List<DiscordApplicationCommandOption>();
             foreach (var parameter in parameters)
             {
                 // Retrieves the attribute
-                var optionAttribute = parameter.GetCustomAttribute<OptionAttribute>() ?? throw new ArgumentException($"Argument {parameter.Name} on method {this.GetFullname(parameter.Member)} requires an {this.GetFullname(typeof(OptionAttribute))}!");
+                var optionAttribute = parameter.GetCustomAttribute<OptionAttribute>() ?? throw new ArgumentException($"Argument \"{parameter.Name}\" on method \"{this.GetFullname(parameter.Member)}\" requires an {this.GetFullname(typeof(OptionAttribute))}!");
 
-                // Remove the type nullability for retrieving the option type from the dictionary.
-                var parameterType = Nullable.GetUnderlyingType(parameter.ParameterType) ?? parameter.ParameterType;
-                if (!_validOptionTypes.TryGetValue(parameterType, out var parameterOptionType))
+                Type parameterType;
+                if (parameter.ParameterType.IsArray)
                 {
-                    // `string.Join` to prevent hardcoding the types. Could lead to extremely an long error message when custom converters are implemented.
-                    throw new ArgumentException($"Argument {parameter.Name} on method {this.GetFullname(parameter.Member)} has an invalid type! Acceptable types are: {string.Join(", ", _validOptionTypes.Keys.Select(type => type.Name))}");
+                    parameterType = parameter.ParameterType.GetElementType();
```

Comment:
only allow this if `params` is specified. otherwise, converters could be allowed to handle arrays by themselves, in which case we shouldn't try to get the element type unconditionally

Reply:
iirc an exception is thrown when an array is specified but params isn't used, so we shouldn't have to worry about it during this code. I'll add an extra check regardless